src/patch-global-cjs.ts

```diff
@@ -0,0 +1,5 @@
+import { URLPattern } from "./index.impl.cjs";
```

Comment:
Nit: you can probably just have one version of this file that imports a source file and the import should be compiled by esbuild to point to the right distribution file. Requires some build config changes.

Reply:
Yeah, I gave it a try and couldn't find a easy way to do it with esbuild. The --define stuff doesn't work for strings and seem to be post compile so it would complain about the import statement. I might look at it later.

Thanks for the review!